elastic/apm-server
# systemtest: migrate TLS cert tests to Go

## Motivation/summary

Migrate TLS system tests to Go. Only essentials are migrated, some less important configurations (e.g. optional client certs) are no longer tested.

We now capture net/http server error logs with logp. Previously they were sent to the standard library's log package, which logp disables by default.

See commit message for program used to generate client certificate/key pair.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Run apm-server with `-E apm-server.ssl.client_authentication=required`
2. `curl -k https://localhost:8200`; should fail because no client cert is provided
3. Check apm-server logs for an error message with logger "beater.http" indicating the client did not provide a certificate

## Related issues

None
